lib/Controller/LoginController.php

```diff
@@ -329,50 +329,59 @@ public function code($state = '', $code = '', $scope = '', $error = '', $error_d
 		$email = $payload->{$emailAttribute} ?? null;
 		$quota = $payload->{$quotaAttribute} ?? null;
 
-		$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_UID, $payload, $userId);
-		$this->eventDispatcher->dispatchTyped($event);
-		if (!$event->hasValue()) {
-			return new JSONResponse($payload);
-		}
+		$oidcSystemConfig = $this->config->getSystemValue('user_oidc', []);
+		$autoProvisionAllowed = (!isset($oidcSystemConfig['auto_provision']) || $oidcSystemConfig['auto_provision']);
+		if (!$autoProvisionAllowed) {
+			$user = $this->userManager->get($userId);
+			if ($user === null) {
+				return new JSONResponse(['Failed to provision user']);
+			}
+		} else {
```

Comment:
We could move the whole user provisioning (both if/else branches) to a separate provisionUser method that returns the IUser. Would make the code a bit more separated and could then return really for the auto provision to avoid one nesting level-

Reply:
> and could then return really for the auto provision to avoid one nesting level-

I didn't get that part :grin:

Anyway, yep, done.

Reply:
Damn autocorrection, return **early** of course ;)